frontend/src/components/tabs/InputOutputTab.tsx

```diff
@@ -49,20 +50,44 @@ export function InputOutputTab({ execution, namespace }: IOTabProps) {
   // TODO(jlyaoyuli): Display other information (container, args, image, command)
 
   // Retrieves input and output artifacts from Metadata store.
-  const { isSuccess, error, data } = useQuery<LinkedArtifact[], Error>(
+  const { isSuccess, error, data: linkedArtifacts } = useQuery<LinkedArtifact[], Error>(
     ['execution_artifact', { id: executionId, state: execution.getLastKnownState() }],
     () => getLinkedArtifactsByExecution(execution),
     { staleTime: Infinity },
   );
 
+  const { data: artifactTypes } = useQuery<ArtifactType[], Error>(
+    ['artifact_types', { linkedArtifact: linkedArtifacts }],
+    () => getArtifactTypes(),
+    {},
+  );
+
+  let artifactTypeNames: string[] = [];
```

Comment:
Question: we use this piece of code to get artifact type name in both `InputOutTab` and `RunDetailsV2`. Should we wrap it and put into new or existed `util` file?

https://github.com/kubeflow/pipelines/pull/8093#discussion_r938281022

Reply:
That is a good point, I think it makes sense to consolidate to `util` file.